crates/biome_js_analyze/tests/specs/nursery/noInvalidUseBeforeDeclaration/invalid.js.snap

```diff
@@ -0,0 +1,177 @@
+---
+source: crates/biome_js_analyze/tests/spec_tests.rs
+expression: invalid.js
+---
+# Input
+```js
+a; const a = 0;
+const b = b;
+c; let c;
+let d = d;
+e; var e;
+var f = f;
+function f(a = b, b = 0) {}
+function g(a = a) {}
+```
+
+# Diagnostics
+```
+invalid.js:1:1 lint/nursery/noInvalidUseBeforeDeclaration ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
+
+  ! This variable is used before its declaration.
+  
+  > 1 │ a; const a = 0;
+      │ ^
+    2 │ const b = b;
+    3 │ c; let c;
+  
+  i The variable is declared here:
+  
+  > 1 │ a; const a = 0;
```

Comment:
I would expect to highlight the `const a`, not the assignment

Reply:
I changed the code to highlight the identifier. I didn't highlight the kind, because several declarators can be in the same declaration. e.g. `const a = 0, b = 0;`.